File: arrays-hashing/217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False
ans = Solution()
print(ans.containsDuplicate([1,1,1,3,3,4,3,2,4,2]))

File: arrays-hashing/217. Contains Duplicate.java
import java.util.*;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (Integer i : nums) {
            if (seen.contains(i)) {
                return true;
            }
            seen.add(i);
        }
        return false;
    }
}
